### PR TITLE
KF-1750 api server rock

### PR DESCRIPTION
**Pull request by misohu**
Docker file: https://github.com/kubeflow/pipelines/blob/2.0.0-alpha.7/backend/Dockerfile

### CODE REVIEW

File: api-server/rockcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,116 @@
+name: kfp-api
+summary: An image for using Kubeflow pipelines API
+description: An image for using Kubeflow pipelines API
+version: "2.0.0-alpha.7_22.04_1"
+base: ubuntu:20.04
```
Comment (i-chvets) [inline api-server/rockcraft.yaml:5]: Should base be 22.04?

File: api-server/rockcraft.yaml [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+license: Apache-2.0
+platforms:
+  amd64:
+
+services:
+  base-admission-webhook:
```
Comment (i-chvets) [inline api-server/rockcraft.yaml:11]: Name of the service is not correct.

File: api-server/rockcraft.yaml [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    override: merge
```
Comment (i-chvets) [inline api-server/rockcraft.yaml:12]: Add summary:
```
  summary: "pipelines api-server service"
```

File: api-server/rockcraft.yaml (hunk 1 above)
Comment (misohu) [reply api-server/rockcraft.yaml:5]: Keeping 20.04 for now for some reason I cant install python 3.7 in part on 22.04

File: api-server/rockcraft.yaml [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+      ./hack/install-go-licenses.sh
+      $GOBIN/go-licenses check ./backend/src/apiserver
+      $GOBIN/go-licenses csv ./backend/src/apiserver > $CRAFT_STAGE/licenses.csv && \
+      diff $CRAFT_STAGE/licenses.csv backend/third_party_licenses/apiserver.csv && \
+      $GOBIN/go-licenses save ./backend/src/apiserver --save_path $CRAFT_STAGE/NOTICES
+    build-packages:
+      - cmake
+      - clang
+      - musl-dev
+      - openssl
```
Comment (jnsgruk) [inline api-server/rockcraft.yaml:46]: Nit but I'd shift this above the `override-build` alongside other deps and setup

File: api-server/rockcraft.yaml [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+      - jq
+      - wget
+    override-build: |
+      curl -sLO https://github.com/argoproj/argo-workflows/releases/download/${ARGO_VERSION}/argo-linux-amd64.gz
+      gunzip argo-linux-amd64.gz
+      chmod +x argo-linux-amd64
+      mv ./argo-linux-amd64 $CRAFT_PART_INSTALL/argo
+      cp ./backend/src/apiserver/config/sample_config.json ./samples/
+      wget https://bootstrap.pypa.io/get-pip.py && python3 get-pip.py
+      pip install --upgrade pip
```
Comment (jnsgruk) [inline api-server/rockcraft.yaml:70]: Why not just add `python3-pip` to the `build-packages`?

File: api-server/rockcraft.yaml [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+      python3 -m pip install -r requirements.txt --no-cache-dir
+      set -e; \
+      < ./samples/sample_config.json jq .[].file --raw-output | while read pipeline_yaml; do \
+      pipeline_py=".${pipeline_yaml%.yaml}"; \
+      mode=`< ./samples/sample_config.json jq ".[] | select(.file == \".${pipeline_yaml}\") | (if .mode == null then \"V1\" else .mode end)" --raw-output`; \
+      mv "$pipeline_py" "${pipeline_py}.tmp"; \
+      echo 'import kfp; kfp.components.default_base_image_or_builder="gcr.io/google-appengine/python:2020-03-31-141326"' | cat - "${pipeline_py}.tmp" > "$pipeline_py"; \
+      dsl-compile --py "$pipeline_py" --output ".$pipeline_yaml" --mode "$mode" || python3 "$pipeline_py"; \
+      done
```
Comment (jnsgruk) [inline api-server/rockcraft.yaml:79]: Not urgent but this is pretty tough going to read, can this be restructured?

Comment (misohu) [reply api-server/rockcraft.yaml:78]: Just to stay consistent with the upstream Dockerfile we may want to keep it similar maybe

**Pull request merged**